### PR TITLE
Generative NFTs

### PR DESCRIPTION
**Pull request by christos-h**
## Motivation

We would like an example showcasing creative applications combining NFTs and generative models.

## Proposal

`gen-nft` is an application which uses the Tiny Llama story model to create unique NFTs from prompts.

image

## Test Plan

Tested manually.

### CODE REVIEW

**Review by andresilva91 — COMMENTED** [review]
Do you have a screenshot or video of the UI?

**Comment by christos-h** [comment]
> Do you have a screenshot or video of the UI?

See description @andresilva91 .

File: examples/gen-nft/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        )
+    }
+```
+
+### Using Web Frontend
+
+Installing and starting the web server:
+
+```bash
+cd examples/non-fungible/web-frontend
```
Comment (ma2bd) [inline examples/gen-nft/README.md:141]: `gen-nft` ?

File: examples/gen-nft/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+export PATH="$PWD/target/debug:$PATH"
+source /dev/stdin <<<"$(linera net helper 2>/dev/null)"
+
+linera_spawn_and_read_wallet_variables linera net up --testing-prng-seed 37
+```
+
+Compile the `non-fungible` application WebAssembly binaries, and publish them as an application bytecode:
+
+```bash
+(cd examples/non-fungible && cargo build --release)
```
Comment (ma2bd) [inline examples/gen-nft/README.md:45]: `gen-nft`

**Comment by ma2bd** [comment]
@christos-h Is the new README file run in CI? I believe you must add it to `linera-service/tests/readme_test.rs`.

File: examples/gen-nft/web-frontend/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# Getting Started with the Non Fungible App
```
Comment (afck) [inline examples/gen-nft/web-frontend/README.md:1]:
```suggestion
# Getting Started with the Gen-NFT App
```

**Review by afck — APPROVED** [review]
Looks good to me, but I only glanced over the parts that I suspect differ from `non-fungible` and the `llm` example.

**Pull request merged**